Expensify/App
# Serialize and store requests made while offline to re-queue when opening a new browser session

### Details
This PR serializes `Report_AddComment` requests so that they persist in Onyx when we are offline. This does not yet work with attachments and we may eventually need some kind of solution to manage persisting attachments - but not yet.

Also in this PR, I simplified the offline detection to rely on `NetInfo` only. I thoroughly tested `NetInfo` on all platforms and couldn't find any scenarios where it didn't perform well. The only issue is that offline features on native require "real" devices to test. I did discover that `NetInfo` wasn't working correctly in Electron so there is a fix in here for that too.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/155359

### Tests
**Testing Caveats**

Testing `NetInfo` offline features on simulators and emulators doesn't seem to work. A real device + ngrok or production API should be used instead.

1. Go offline by moving to Airplane mode or checking the "offline" setting in Chrome dev tools network tab
2. Attempt to leave a comment on a report (with no attachments)
3. Attempt to leave an attachment on a report
3. Close tab or force quit the app
4. Reopen the app
5. Verify the comment appears on the report but the attachment does not appear

### Tested On

- [x] iOS (works great on my iPad mini)

### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
